### PR TITLE
Add first public version

### PR DESCRIPTION
**Pull request by JinneeJ**
This PR adds first public version of plugin. The main goal of plugin is to run [Ktlint](https://github.com/pinterest/ktlint) for changed `kt` files (more details in README.md). This PR doesn't contain any notable changes in implementation compared to internal version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+    // Name of branch used in tasks `ktlint*ChangesFrom*` to compare with current branch. Should be specified without `origin`, though plugin will use remote branch.
+    // Default: main.
+    mainBranch = "..."
+    // Ktlint version.
+    // Required and doesn't have default value.
+    version = "..."
+}
+```
+
+Also see plugin page in [Gradle Plugin Portal](https://plugins.gradle.org/plugin/com.doist.gradle.ktlint)
```
Comment (pedrodanielcsantos) [inline README.md]: Nit: missing a dot at the end. 😊

File: README.md [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+Run any gradle task from this plugin on CI or locally.
+
+### Tasks
+
+- `ktlintCheck` - Checks code styles for all kotlin files in the project.
+- `ktlintFormat` - Formats all kotlin files in the project.
+- `ktlintCheckUncommittedChanges` - Checks code styles for all changed kotlin files.
+- `ktlintFormatUncommittedChanges` - Formats code style for all changed kotlin files.
+- `ktlintCheckChangesFrom*` - Checks code style for all changed `kt` files against main branch.
+- `ktlintFormatChangesFrom*` - Formats code styles for all changed `kt` files against main branch.
```
Comment (pedrodanielcsantos) [inline README.md]: The `*` at the end is a wildcard for any branch name, right? If it is, then we should update the description. If it's not, then we should remove the wildcard from the task name.

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    isAutomatedPublishing = true
+}
+
+pluginBundle {
+    website = "https://github.com/Doist/ktlint-gradle-plugin"
+    vcsUrl = "https://github.com/Doist/ktlint-gradle-plugin.git"
+
+    plugins.getByName(pluginName) {
+        displayName = "Ktlint Gradle Plugin"
+        description = "This plugin is mainly designed to check code styles on changed part of the code"
```
Comment (pedrodanielcsantos) [inline build.gradle.kts]: Suggestion: _"This plugin is designed to validate the code style of changed kt files."_

**Review by pedrodanielcsantos — APPROVED** [review]
Awesome work @JinneeJ, great to see this work going public! 👏
Only left a couple of nits, other than that, looks good to me. 🚀

**Comment by JinneeJ** [comment]
Thanks for the review and suggestions! Updated and will merge :)

**Pull request merged**